File: src/main/java/commands/slashadapters/adapters/FisheryManageAdapter.java
package commands.slashadapters.adapters;

import commands.runnables.fisherysettingscategory.FisheryManageCommand;
import commands.slashadapters.Slash;
import commands.slashadapters.SlashAdapter;
import commands.slashadapters.SlashMeta;
import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
import net.dv8tion.jda.api.interactions.commands.OptionType;
import net.dv8tion.jda.api.interactions.commands.build.CommandData;
import net.dv8tion.jda.api.interactions.commands.build.OptionData;

@Slash(command = FisheryManageCommand.class)
public class FisheryManageAdapter extends SlashAdapter {

    public CommandData addOptions(CommandData commandData) {
        String[] components = new String[]{ "fish", "coins", "dailystreak", "fishing_rod", "fishing_robot", "fishing_net", "metal_detector", "role", "survey", "work", "reset"};
        OptionData optionData = new OptionData(OptionType.STRING, "component", "Which component should be modified?", false);
        for (String component : components) {
            optionData.addChoice(component, component);
        }

        return commandData
                .addOption(OptionType.USER, "member", "Select a member", false)
                .addOption(OptionType.USER, "member2", "Select a member", false)
                .addOption(OptionType.USER, "member3", "Select a member", false)
                .addOption(OptionType.USER, "member4", "Select a member", false)
                .addOption(OptionType.USER, "member5", "Select a member", false)
                .addOption(OptionType.STRING, "member_id", "Select a member", false)
                .addOption(OptionType.ROLE, "role", "Select all members of a role", false)
                .addOptions(optionData)
                .addOption(OptionType.STRING, "operation", "What operation should be performed on the component? (e.g. +1, -4, 6)");
    }

    @Override
    public SlashMeta process(SlashCommandEvent event) {
        return new SlashMeta(FisheryManageCommand.class, collectArgs(event));
    }

}

File: src/main/java/commands/slashadapters/adapters/FisheryManageAdapter.kt
package commands.slashadapters.adapters

import commands.Category
import commands.runnables.fisherysettingscategory.FisheryManageCommand
import commands.slashadapters.Slash
import commands.slashadapters.SlashAdapter
import commands.slashadapters.SlashMeta
import constants.Language
import core.TextManager
import mysql.hibernate.entity.guild.GuildEntity
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.dv8tion.jda.api.interactions.commands.Command.Choice
import net.dv8tion.jda.api.interactions.commands.OptionType
import net.dv8tion.jda.api.interactions.commands.build.SlashCommandData

@Slash(command = FisheryManageCommand::class)
class FisheryManageAdapter : SlashAdapter() {

    public override fun addOptions(commandData: SlashCommandData): SlashCommandData {
        val components =
            arrayOf("fish", "coins", "dailystreak", "fishing_rod", "fishing_robot", "fishing_net", "metal_detector", "role", "survey", "work", "reset")
        val optionData = generateOptionData(OptionType.STRING, "component", "fisherymanage_component", false)
        components.forEachIndexed { i, component ->
            var choice: Choice? = null
            Language.values().forEach { language ->
                val name = when (i) {
                    in 0..2 -> TextManager.getString(language.locale, Category.FISHERY_SETTINGS, "fisherymanage_options").split("\n")[i]
                    in 3..9 -> TextManager.getString(language.locale, Category.FISHERY, "buy_product_${i - 3}_0")
                    10 -> TextManager.getString(language.locale, Category.FISHERY_SETTINGS, "fisherymanage_state0_reset")
                    else -> throw IndexOutOfBoundsException()
                }
                if (language == Language.EN) {
                    choice = Choice(name, component)
                } else {
                    choice!!.setNameLocalization(language.discordLocales[0], name)
                }
            }
            optionData.addChoices(choice)
        }
        return commandData.addOptions(
            generateOptionData(OptionType.STRING, "members", "fisheryset_members", false),
            generateOptionData(OptionType.STRING, "roles", "fisherymanage_mentionroles", false),
            optionData,
            generateOptionData(OptionType.STRING, "operation", "fisherymanage_operation")
        )
    }

    override fun process(event: SlashCommandInteractionEvent, guildEntity: GuildEntity): SlashMeta {
        return SlashMeta(FisheryManageCommand::class.java, collectArgs(event))
    }

}